Makes combi-knife actually function and fixes its tools.

BladeburstNINJA opened the pull request:
Widens target choice so that combi-knife tools can actually function on most objects.

<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

:cl: BladeburstNINJA
tweak: Makes combi-knife functional
/:cl:
Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You'll find a README and example file in .\html\changelogs\ for further instructions.

You can also find a template for adding your changelog directly to the PR description here: https://github.com/Baystation12/Baystation12/wiki/Automatic-changelog-generation
-->

BladeburstNINJA committed:
Update swiss.dm

Widens target choice so that combi-knife tools can actually function on most objects.

Jackofashepard commented:
Doesnt this kill the point of toolboxes and belts?

BladeburstNINJA commented:
No, because you can already get an augment in your arm that enables you to have all the tools without a need for any type of storage. This only grants you three of the tools in your tool belt and still requires you to find storage for it.

OolongCow commented:
> Doesnt this kill the point of toolboxes and belts?

If we're being honest, here, the only tools that're at all difficult to find and are still useful are multitools (only a few set spawns, rare spawn in maint), screwdrivers (very specific toolboxes from maint or vendors) or welders (maint/emergency toolbox loot or vendors). And even then, there's a vendor that dispenses 8 - 10 of them, minus the toolbelt and multitool, one z-level away from the majority of the crew's spawn point. All this really does is make some basic tools that might be useful not require you metarush tool storage at roundstart.

The combination of tools it offers is also basically worthless, since you have to go get tools regardless for any meaningful work. The only unique and good tool omnitools have is Med Tech's glass breaker, and even that's pretty niche.

Jackofashepard commented:
I'm hesitant to merge this since holding most of the tools thats in this swiss knife usually means sacrificing your belt slot for a toolbelt or a fair amount of space inside your backpack, and considering the things you can do with tools its a pretty good trade off

BladeburstNINJA commented:
So your main issue with this seems to be, storage space and that your essentially collapsing the size of three tools into one tool (Correct me if I'm wrong). The problem with this is that there are already other ways to minimize the impact tools have on your storage, this would only be an additional way to do it. As of current, let's say you don't need all the tools, just five. You fit five tools into a webbing vest, without sacrificing any belt or backpack storage. Let's say you need all the tools, you can get an arm augment that grants you all of the tools (minus cable coil) for a cost of ZERO storage space. Storage space should not be an issue because there are already workarounds to it.

Jackofashepard commented:
If you use a webbing you sacrifice your accessory slot for tools, if you take the implant you sacrifice a organic arm which comes with its own drawbacks, basically I'm worried since this genuinely has 0 drawbacks.

BladeburstNINJA commented:
But it doesn't have zero drawbacks. If you lose your combi-knife you just lost three tools in one. Normally if you lose one tool, you only have to replace that one tool.

BladeburstNINJA commented:
Also, you could argue that a mechanical arm is better than an organic one.

RenGusta commented:
Just make the tool times like triple the length like ???

Ez Balance

Jackofashepard commented:
Right so i consulted some people, i'll merge it if you are able to make the swiss tools a poorer choice to actual tools, i.e. make them work slower

Jackofashepard commented:
Also keep the check on at least the swiss crowbar, it'd be a bit comical to pry open a airlock with a finger size crowbar

BladeburstNINJA commented:
Sounds good, I'll need a bit to work on the code.

BladeburstNINJA committed:
Combi-knife rebalance

Crowbar no longer works on things such as apcs, doors, and foor tiles. Tools now have a time delay and take 3 seconds to use.

Jackofashepard commented on code/game/objects/items/weapons/material/swiss.dm:
```diff
@@ -113,12 +113,18 @@
-			can_use_tools = TRUE
-			. = ..()
-			can_use_tools = FALSE
+	if(istype(target, /obj))
```
Check for target size

BladeburstNINJA replied:
for what reason? If it checks for target size you would be unable to use the screwdriver for example on things like a door or window, wouldn't you?
